### ISSUE TITLE
Change is_external() function in MenuItem.php to respect "Open in new tab" checkbox

### ISSUE THREAD

**Issue by gserafini** [timber/timber#1629]
### Expected behavior
If a user clicks the "Open in new tab" checkbox in Dashboard > Appearance > Menus, link should be opened in a new tab, regardless of if it in the same domain or subdomain or not, if the helper `is_external` function code is used.

### Actual behavior
Menu links that have "Open in new tab" do not open in a new tab.  The `is_external` function is hard-coded to return false in the case of a subdomain, and true for all other domains that are different from the base domain.

### Steps to reproduce behavior
1. Open a menu in Dashboard > Appearance > Menus
2. Add a link
3. Click the "Open in new tab" checkbox (to view this, make sure that Targets is selected in the Screen Options
4. Load page that includes a Timber navigation menu with the suggested usage code: `<a href="{{ item.link }}" target="{{ item.is_external ? '_blank' : '_self' }}">`
5. Click the link that you specified to open in a new tab.  If it is an internal or subdomain link it will open in the same window, and external custom URLs (excluding subdomains) will always open in a new window, regardless of the WordPress setting for that link.

### Suggested solution
Update the is_external() function call in lib/MenuItem.php to check for the existence of the checkbox preference and respect it if it's set.  Get rid of old behavior that assumes that all non-subdomain links should be opened in a new window.  Or... make a new function that does this below, maybe called is_new_tab() or something.

~Line 282 in MenuItem.php:

```
	public function is_external() {
		if ( isset ( $this->_menu_item_target ) && $this->_menu_item_target ==  "_blank") {
			return true;
		}
		return false;
	}
```

### What version of WordPress, PHP and Timber are you using?
WordPress 4.9.1 (latest), PHP 7.0.26, Timber 1.6.0

### How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)
Upgraded to newest version via plugin updater in WordPress dashboard.

**Comment by gchtr** [comment]
https://github.com/timber/timber/issues/1629

Hey @gserafini, thanks a lot for taking the time to properly fill out the issue template!

I very much like your idea to respect the menus setting for opening a menu item in a new tab.

I think that the idea behind `is_external()` is really to tell whether the link is to an external site (other domain or subdomain). It might for example also be used to display an icon next to the link for better accessibility. So I’d go for your suggested alternative solution to use another function name for your idea.

Naming it `is_new_tab()` is not accurate enough in my opinion, because whether a link opens in a new tab depends on the browser setting of the user. It could also open in a new window. In the default WordPress menu walker, `target="_blank"` is added when that checkbox is ticked, and the option value is actually `_blank` if it is set, so my suggestion would be to use `is_target_blank()` as a function name:

```twig
<a href="{{ item.link }}" target="{{ item.is_target_blank ? '_blank' : '_self' }}">
```

Or 

```twig
<a href="{{ item.link }}" {{ item.is_target_blank ? 'target="_blank"' }}>
```

We could also introduce a new `target()` method in the MenuItem class, that would set the proper target based on the setting in the backend?

```twig
<a href="{{ item.link }}" target="{{ item.target }}">
```

However, when target is `_self`, it can also be omitted, so I’d rather have no target than having `_self` as a value. So I’m still a little hesitant about adding a `target` method.

What do you think?

**Comment by gserafini** [comment]
Right now in the standard WordPress Appearance > Menus system there's only the option to check a box: "Open link in new tab".  I'd say let's go with adding the `is_target_blank()` method as you suggest, since the potential new `target()` method you mention is either going to be _blank or _self, depending on if that checkbox is checked or not.

This approach seems like it makes the most sense to me for suggested usage:

```
<a href="{{ item.link }}" {{ item.is_target_blank ? 'target="_blank"' }}>
```

**Comment by gserafini** [comment]
Thanks for the work on this one! :)

**Comment by jarednova** [comment]
Closed by #1629 and #1701

**Comment by jarednova** [comment]
I mean closed by #1701 and #1715 !

**Comment by diaboloshogunate** [comment]
Can this be added to the documentation?

**Comment by gchtr** [comment]
@diaboloshogunate The feature is not available in the current version of Timber. It will be available with Timber 2.0. It’s already documented, but you will only see it in the documentation when version 2.0 will be released.
